File: backend/student-manager/src/main/java/com/student_manager/feature/student/StudentService.java
package com.student_manager.feature.student;

import java.util.List;

public interface StudentService {
    StudentDTO findById(Long id);

    /**
     * Resolves the student record linked to an authenticated account, matching
     * the account's e-mail to {@link Student#getEmail()}. Used by {@code GET
     * /api/students/me} so a STUDENT can see their own record without being able
     * to read the whole roster.
     */
    StudentDTO findByAccountUsername(String username);

    /**
     * True when the account identified by {@code username} is linked (by e-mail)
     * to the student row {@code studentId}. Used by authorization checks so a
     * STUDENT can only reach their own data. Never throws — a missing account or
     * student row simply yields {@code false}.
     */
    boolean accountOwnsStudent(String username, Long studentId);

    List<StudentDTO> findAll();
    StudentDTO create(CreateStudentRequest request);
    StudentDTO update(Long id, CreateStudentRequest request);
    void delete(Long id);
}

File: frontend/src/services/studentService.ts
import api from './api';
import type { Student, CreateStudentRequest } from '../types';
import type { AxiosResponse } from 'axios';

const studentService = {
  getAll: (): Promise<AxiosResponse<Student[]>> =>
    api.get<Student[]>('/students'),

  getById: (id: number): Promise<AxiosResponse<Student>> =>
    api.get<Student>(`/students/${id}`),

  getMe: (): Promise<AxiosResponse<Student>> =>
    api.get<Student>('/students/me'),

  create: (data: CreateStudentRequest): Promise<AxiosResponse<Student>> =>
    api.post<Student>('/students', data),

  update: (id: number, data: CreateStudentRequest): Promise<AxiosResponse<Student>> =>
    api.put<Student>(`/students/${id}`, data),

  delete: (id: number): Promise<AxiosResponse<void>> =>
    api.delete(`/students/${id}`),
};

export default studentService;
